administrator/components/com_content/src/Model/ArticleModel.php

```diff
@@ -751,6 +751,12 @@ public function save($data)
             } elseif ($data['alias'] == $origTable->alias) {
                 $data['alias'] = '';
             }
+
+            if (empty($this->workflow)) {
```

Comment:
```suggestion
            if (!$this->workflowEnabled) {
```
From my quick look at the code, `$this->workflow` is never empty, so the code is not right. We should change to if `(!$this->workflowEnabled)` instead. No clue why do you need to set `$data['transition'] = 1;` in the else, I will try to look at it later.

Reply:
> No clue why do you need to set $data['transition'] = 1; in the else, I will try to look at it later.

This "else" is the only reason this PR exists.

Reply:
I would need to look at the workflow code at a later time, but my first thought is that we only need to set state = 0 on Save As Copy if workflow is not enabled, so I think only the `if` is enough.

Reply:
Well, no, that's the code that existed, caused the issue and was removed.

Please read the PR description and the linked issues and previous PRs.

If you are suggesting setting transition = 1 is not the right way to trigger the workflow unpublish transition, that's quite possible, it's the first and only way I found, by looking at what's done when one uses workflow transition selector.

Reply:
By looking at the original PR, it was reverted because it breaks workflow. With the suggested if clause, it will only set state = 0 if workflow is disabled, and leave everything else works as how it is at the moment, so it should work

Anyway, at least your if clause is not right, because as I said, `$this->workflow` is always an object, it never empty, so the code inside if won't be executed. 

I do not know about the `else` yet until I can look at the whole workflow code.

Reply:
>  and leave everything else works as how it is at the moment,

Euh no. The addition is data['transition'] = 1; which triggers the workflow transition to unpublished state.

> $this->workflow is always an object, it never empty, so the code inside if won't be executed.

Of course it will be executed,  if $this->workflow is false or non-existing. 

empty() is the safe way to check if something evaluates to false. empty does not check for emptyness. It replaces isset($var) && !$var (which surprisingly can still be found in a few places in Joomla)

I give you that as the model uses the workflow trait, workflowEnabled is certain to be defined, so empty() is not needed here, it could be replaced with just !$this-workflowEnabled

Reply:
But thanks for pointing out that I am using the wrong property name (workflow instead of workflowEnabled). Bad quick change yesterday.

Reply:
> empty() is the safe way to check if something evaluates to false. empty does not check for emptyness. It replaces isset($var) && !$var (which surprisingly can still be found in a few places in Joomla)

I know that PHP basic :).

> I give you that as the model uses the workflow trait, workflowEnabled is certain to be defined, so empty() is not needed here, it could be replaced with just !$this-workflowEnabled

That's what I suggested. I suggested you to use **if (!$this->workflowEnabled) {**, but your code still calls empty function which is not necessary. Please update your code to not use empty function.

Reply:
> That's what I suggested. 

That's not what you said or suggested. You said:

> Anyway, at least your if clause is not right, because as I said, $this->workflow is always an object, it never empty, so the code inside if won't be executed.

Reply:
Please check my first comment :), and see the attached screenshot, I made a code suggestion, you could just commit that suggestion but you modified it yourself :)

![code_suggestion](https://github.com/user-attachments/assets/72784864-7c99-4016-8a16-1b47ed78675a)

Reply:
Probably a personal difference here, I barely looked at the code but I did pay attention to the comment, which was not really relevant and therefore did not go any further (but did which is when I realized I was using the wrong variable name in that PR). 

As empty($var) and !$var are strictly equivalent here, I did not understand why you said later "it never empty, so the code inside if won't be executed.", which of course was incorrect.

Anyway, I can see testers are testing so I'll focus on that now. That issue is so frustrating in Joomla 5+ that I do intend to fix it now.

Reply:
> I did not understand why you said later "it never empty, so the code inside if won't be executed."

I said that because your original code use `if (empty($this->workflow))`, and `$this->workflow` is initialized here https://github.com/joomla/joomla-cms/blob/5.2-dev/libraries/src/MVC/Model/WorkflowBehaviorTrait.php#L90, it is always an object, and that is the reason I said the code inside if will never be executed, and I am correct.

You should still remove the usage if empty function, no point of calling a function when it is not needed like that. But I will stop commenting here and leave it to you.

Reply:
I get it now. No $this->workflow was a typo, as I made this PR from Gtihub directly and don't have a local dev setup.